spec/unit/puppet/provider/rabbitmq_plugin/rabbitmqctl_spec.rb

```diff
@@ -9,17 +9,84 @@
   end
   let(:provider) { provider_class.new(resource) }
 
-  it 'matches plugins' do
-    provider.expects(:rabbitmqplugins).with('list', '-E', '-m').returns("foo\n")
-    expect(provider.exists?).to eq(true)
-  end
-
   it 'calls rabbitmqplugins to enable when node not running' do
     provider.class.expects(:rabbitmq_running).returns false
     provider.expects(:rabbitmqplugins).with('enable', 'foo')
     provider.create
   end
 
+  describe '#instances' do
+    it 'exists' do
+      expect(provider_class).to respond_to :instances
+    end
+
+    # rubocop:disable RSpec/MultipleExpectation
+    it 'retrieves instances' do
+      provider.class.expects(:plugin_list).returns(%w[foo bar])
+      instances = provider_class.instances
+      expect(instances.size).to eq(2)
```

Comment:
Is this expectation really that useful? If it's not 2, it's hard to debug because you don't get the actual values. Below you compare names so you at least know more.

Reply:
Agreed and fixed!